subsys/net/lib/sockets/sockets.c

```diff
@@ -419,6 +420,23 @@ int zsock_accept_ctx(struct net_context *parent, struct sockaddr *addr,
 		return -1;
 	}
 
+	/* Check if the connection is already disconnected */
+	last_pkt = k_fifo_peek_tail(&ctx->recv_q);
+	if (last_pkt) {
+		if (net_pkt_eof(last_pkt)) {
+			sock_set_eof(ctx);
+			errno = ECONNABORTED;
```

Comment:
I don't think this is the right handling, based on https://github.com/zephyrproject-rtos/zephyr/pull/19919#issuecomment-546262916